test/rpc/request_test.exs

```diff
@@ -0,0 +1,96 @@
+defmodule Soroban.RPC.Client.CannedRequestImpl do
+  @moduledoc false
+
+  @behaviour Soroban.RPC.Client.Spec
+
+  @impl true
+  def request(
+        _endpoint,
+        _url,
+        _headers,
+        _params,
+        _opts
+      ) do
+    send(self(), {:soroban_response, 200})
+    {:ok, %{result: "result"}}
+  end
+end
+
+defmodule Soroban.RPC.RequestTest do
+  use ExUnit.Case
+
+  alias Soroban.RPC.{Client.CannedRequestImpl, GetTransactionResponse, HTTPError, Request}
+  alias Soroban.Test.Fixtures.RPC
+
+  setup do
+    Application.put_env(:soroban, :http_client_impl, CannedRequestImpl)
+
+    on_exit(fn ->
+      Application.delete_env(:soroban, :http_client_impl)
+    end)
+
+    url = "https://rpc-futurenet.stellar.org:443/"
+    headers = [{"Content-Type", "application/json"}]
+    endpoint = "getTransaction"
+
+    params = %{
+      hash: "2a3c55cc19ed98e40f8eff426f1e16f2cff84bb0a75be45c2ed4b62159380e9a"
+    }
```

Comment:
Guys let's accept the params as keyword list instead of map.

IMO The pipeline looks better with keyword list:
```elixir
endpoint
  |> Request.new()
  # look at here
  |> Request.add_params(param1: "2a3c5...", param2: "value...", another_param: "value...")
  |> Request.add_headers(headers)
  |> Request.perform()
  |> Request.results(as: GetTransactionResponse)
```